native/src/main/java/io/ballerina/stdlib/http/api/client/endpoint/CreateSimpleHttpClient.java

```diff
@@ -55,7 +56,17 @@ public static Object createSimpleHttpClient(BObject httpClient, BMap globalPoolC
             if (!urlString.strip().isEmpty()) {
                 String[] urlStrings = urlString.split(HttpConstants.SCHEME_SEPARATOR, 2);
                 if (urlStrings.length == 1) {
-                    urlStrings = new String[]{HttpConstants.HTTP_SCHEME, urlStrings[0]};
+                    if (isClientSecurityConfigured(clientEndpointConfig)) {
+                        urlStrings = new String[]{HttpConstants.HTTPS_SCHEME, urlStrings[0]};
```

Comment:
Can use ternary operator. Not mandatory
```
urlStrings = isClientSecurityConfigured(clientEndpointConfig) ? new String[]{HTTPS_SCHEME, urlStrings[0]} :
         new String[]{HTTP_SCHEME, urlStrings[0]};
```

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-http/pull/962/commits/64c1c41901c3cb77f3a82904dd2452b545b26437